src/main/groovy/io/jenkins/infra/repository_permissions_updater/ArtifactoryAPI.groovy

```diff
@@ -218,17 +219,17 @@ abstract class ArtifactoryAPI {
         @Override
         @CheckForNull String generateTokenForGroup(String username, String group, String description, long expiresInSeconds) {
             withConnection('POST', ARTIFACTORY_TOKEN_API_URL) {
-                setRequestProperty('Content-Type', 'application/x-www-form-urlencoded')
+                setRequestProperty('Content-Type', 'application/json')
```

Comment:
Could you point to any docs that this would work? Or did you try it and it works? Because the docs are pretty clear that the API expects 

> Content-Type: application/x-www-form-urlencoded 

and only

> Produces: application/json

Reply:
I linked to the docs: https://www.jfrog.com/confluence/display/JFROG/Artifactory+REST+API#ArtifactoryRESTAPI-CreateToken.1

Reply:
Wrong API. We're still on https://www.jfrog.com/confluence/display/JFROG/Artifactory+REST+API#ArtifactoryRESTAPI-CreateToken

Reply:
Your looking at the deprecated endpoint:

This API is being deprecated and replaced by the [AccessTokens](https://www.jfrog.com/confluence/display/JFROG/Artifactory+REST+API#ArtifactoryRESTAPI-AccessTokens)

Maybe that's why we are having isuses.

Reply:
> deprecated endpoint

I know. Again, the new one has only existed for a year. This repo and tool have existed for much longer.

> Maybe that's why we are having isuses.

Or maybe, and I know this is a crazy suggestion, Artifactory has always had undocumented length limitations in its API.

Reply:
Reverted to the deprecated API endpoint

Reply:
Do we have an way to spin up artifactory in a docker container to simulate that part?

The easy part is github creating a test repository and only saying it has CD enabled.